features/step_definitions/survey_steps.rb

```diff
@@ -142,7 +142,11 @@
   end
 end
 
-Given('I see the summary {string} with') do |summary, table|
+When('I open a survey link for UPRN {int}') do |uprn|
+  visit root_path + "?uprn=#{uprn}"
```

Comment:
Use `visit "/?uprn=#{uprn}"` - I prefer to avoid url helpers in tests so that bugs in routing don't get masked.